### ISSUE TITLE
MetricsLogger 機能の追加

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochimethod#32]
## 概要

任意のキー・値ペアを CSV に記録する汎用メトリクスロガーを追加する.

## 背景・動機

- CNN 訓練時に loss, accuracy などのメトリクスを記録したい
- 毎回 CSV 書き込みのコードを書くのが面倒
- クラッシュ時にデータが失われないよう自動保存したい

## ユースケース

- 機械学習の訓練ログ (epoch, loss, accuracy)
- データ分析の実験記録 (trial, precision, recall, f1)
- 任意の実験ログ

## 使用例

```python
from pochi import Pochi

pochi = Pochi()
ws = pochi.create_workspace("outputs", subdirs=["logs"])

# メトリクスロガーを作成
metrics = pochi.create_metrics_logger(ws.logs)

# 任意のキー・値で記録
for epoch in range(100):
    train_loss = train_one_epoch()
    val_loss, val_acc = validate()
    metrics.log(epoch=epoch, train_loss=train_loss, val_loss=val_loss, val_acc=val_acc)

# → ws.logs/metrics.csv に自動保存
# epoch,train_loss,val_loss,val_acc
# 0,0.523,0.612,0.78
# 1,0.421,0.501,0.82
# ...

# ファイル名指定
metrics = pochi.create_metrics_logger(ws.logs, filename="train_log.csv")

# 画像処理の例
metrics = pochi.create_metrics_logger(ws.logs, filename="quality.csv")
metrics.log(image="001.jpg", psnr=32.5, ssim=0.95)
metrics.log(image="002.jpg", psnr=30.1, ssim=0.92)
```

## 機能要件

- `pochi.create_metrics_logger(log_dir, filename="metrics.csv")` でロガー作成
- `metrics.log(**kwargs)` で任意のキー・値ペアを記録
- 毎回自動保存 (クラッシュ対策)
- カラム順序は登場順で固定
- 標準ライブラリのみ使用 (pandas 不要)

## 実装イメージ

```python
class MetricsLogger:
    def __init__(self, log_dir: Path, filename: str = "metrics.csv") -> None:
        self._log_dir = Path(log_dir)
        self._log_dir.mkdir(parents=True, exist_ok=True)
        self._filepath = self._log_dir / filename
        self._records: list[dict[str, Any]] = []
        self._columns: list[str] = []  # カラム順序を保持

    def log(self, **kwargs: Any) -> None:
        """任意のキー・値ペアを記録."""
        for key in kwargs:
            if key not in self._columns:
                self._columns.append(key)
        self._records.append(kwargs)
        self._save()

    def _save(self) -> None:
        """CSV に保存."""
        with open(self._filepath, "w", newline="", encoding="utf-8") as f:
            writer = csv.DictWriter(f, fieldnames=self._columns)
            writer.writeheader()
            writer.writerows(self._records)

    @property
    def filepath(self) -> Path:
        """保存先パスを取得."""
        return self._filepath

    @property
    def records(self) -> list[dict[str, Any]]:
        """記録済みデータを取得."""
        return self._records.copy()
```

## ディレクトリ構成

```
pochi/
  metrics/
    __init__.py
    interfaces.py    # IMetricsLogger
    logger.py        # MetricsLogger
```

## Branch

`feat/metrics-logger`

## Labels

- enhancement
